cmd/bucket-replication.go

```diff
@@ -769,6 +769,20 @@ func putReplicationOpts(ctx context.Context, sc string, objInfo ObjectInfo) (put
 		// In case of SSE-C objects copy the allowed internal headers as well
 		if !crypto.SSEC.IsEncrypted(objInfo.UserDefined) || !slices.Contains(maps.Keys(validSSEReplicationHeaders), k) {
 			if stringsHasPrefixFold(k, ReservedMetadataPrefixLower) {
+				if strings.EqualFold(k, ReservedMetadataPrefixLower+"crc") {
+					cs := objInfo.decryptChecksums(0)
+					if cs != nil {
```

Comment:
```suggestion
					if cs != nil {
```

superflous not needed just do `for k, v := range cs`